File: guilib-api/src/main/java/net/developertobi/guilib/api/utils/MathUtils.java
package net.developertobi.guilib.api.utils;

import net.developertobi.guilib.api.item.GuiPos;

public final class MathUtils {

    public static GuiPos slotToPosition(int slot, int columns) {
        return GuiPos.of(slot / columns, slot % columns);
    }

    public static int positionToSlot(GuiPos pos, int columns) {
        return pos.row() * columns + pos.column();
    }

    public static GuiPos nextPositionFromSlot(int slot, int columns) {
        return slotToPosition(slot + 1, columns);
    }
}

File: guilib-api/bin/main/net/developertobi/inventorylib/api/utils/MathUtils.kt
package net.developertobi.inventorylib.api.utils

import net.developertobi.inventorylib.api.item.GuiPos

object MathUtils {

    fun slotToPosition(slot: Int, columns: Int): GuiPos {
        return GuiPos.of(slot / columns, slot % columns)
    }

    fun positionToSlot(pos: GuiPos, columns: Int): Int {
        return pos.row * columns + pos.column
    }

    fun nextPositionFromSlot(slot: Int, columns: Int): GuiPos {
        return slotToPosition(slot + 1, columns)
    }

}
